src/main/java/com/robin/hoteless/Controllers/UserController.java

```diff
@@ -0,0 +1,46 @@
+package com.robin.hoteless.Controllers;
+import com.robin.hoteless.Model.User;
+import org.bson.types.ObjectId;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+
+@RestController
+@RequestMapping ("/user")
+public class UserController {
+
+    //@Autowired
```

Comment:
You're not defining a UserRepository here, which is probably the cause of your NPE (Null Pointer Exception); the ``@Autowired`` annotation tells it to look for a pre-defined @Bean (usually in a @Configuration class) to use for this. Otherwise, UserRepository == null here.